src/features/pass-survey/ui/ActivityStepper.tsx

```diff
@@ -136,7 +146,7 @@ function ActivityStepper({
 
     if (canSkip) {
       trackUserAction(userActionCreator.skip());
-    } else {
+    } else if (!shouldSkipNextUserAction) {
```

Comment:
Indeed it looks like workaround. We build BL based on user actions! 
Can we change the solution to another one.
For example: pass ignoreUserActionTrack as the parameter of onNext, and evaluate this value in the onBeforeNext?